BashCode.Test/FunctionalServiceShould.cs

```diff
@@ -0,0 +1,83 @@
+namespace BashCode.Test
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Moq;
+
+    using Xunit;
+
+    public class FunctionalServiceShould
+    {
+        [Fact]
+        public void TriggeredCallBackWithErrorsWhenHaveNotWhateverModels()
+        {
+            Mock<IWhateverAdapter> whateverAdapterMock = new Mock<IWhateverAdapter>();
+            whateverAdapterMock.Setup(adapter => adapter.FindAllByLastModification())
+                               .Returns(Enumerable.Empty<WhateverModel>().ToList);
+            Mock<Next> callbackMock = new Mock<Next>();
```

Comment:
add setup and callback for get collection parameter 
 callbackMock.Setup(callback => callback(It.IsAny<IReadOnlyList<string>>()))
                        .Callback<IReadOnlyList<string>>(messages => { actualMessages = messages.ToList(); });